dfdaemon/config/config.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+
+	"gopkg.in/yaml.v2"
+)
+
+// -----------------------------------------------------------------------------
+// Properties
+
+// NewProperties create a new properties with default values.
+func NewProperties() *Properties {
+	return &Properties{}
+}
+
+// Properties holds all configurable properties of dfdaemon.
+// The default path is '/etc/dragonfly/dfdaemon.yml'
+type Properties struct {
+	Registries []*Registry `yaml:"registries"`
+}
+
+// Load loads properties from config file.
+func (p *Properties) Load(path string) error {
+	if err := p.loadFromYaml(path); err != nil {
+		return err
+	}
+	var tmp []*Registry
+	for _, v := range p.Registries {
+		if v != nil {
+			if err := v.init(); err != nil {
+				return err
+			}
+			tmp = append(tmp, v)
+		}
+	}
+	p.Registries = tmp
+	return nil
+}
+
+func (p *Properties) loadFromYaml(path string) error {
+	yamlFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("read yaml config from %s error: %v", path, err)
+	}
+	err = yaml.Unmarshal(yamlFile, p)
+	if err != nil {
+		return fmt.Errorf("unmarshal yaml error:%v", err)
+	}
+	return nil
+}
+
+// -----------------------------------------------------------------------------
+// Registry
+
+// NewRegistry create and init registry proxy with the giving values.
+func NewRegistry(schema, host, cert, regx string) (*Registry, error) {
+	reg := &Registry{
+		Schema: schema,
+		Host:   host,
+		Cert:   cert,
```

Comment:
This param looks like never be used in this PR?

Reply:
Yes, this param should be provided when remote https host is self-certificated. But now it works fine because that we don't verify the server's certificate. see code: [transport.go#63](https://github.com/dragonflyoss/Dragonfly/blob/master/dfdaemon/handler/transport.go#L63).

I will fix it:
* verify certificate if the `cert` is not empty
* don't verify certificate if the `cert` is empty

Reply:
done